search/indexing_api.py

```diff
@@ -125,6 +129,68 @@ def _delete_item(document_id, doc_type, index):
         pass
 
 
+def serialize_public_enrolled_user(serialized_enrolled_user):
+    """
+    Creates a public serialized version of an enrolled user.
+
+    Args:
+        serialized_enrolled_user(dict):
+            Serialized enrolled user generated by serialize_program_enrolled_user
+
+    Returns:
+        dict:
+            serialized subset of the original document sanitized for public consumption
+    """
+    # filter out grades, courses passed, etc
+    program = dict_with_keys(
+        serialized_enrolled_user['program'],
+        ['id', 'enrollments', 'is_learner', 'total_courses', ]
+    )
+    program['enrollments'] = [
+        dict_with_keys(enrollment, ['course_title', 'semester', ])
+        for enrollment in program['enrollments']
+    ]
+    # filter out private profile information
+    profile = dict_with_keys(
+        serialized_enrolled_user['profile'],
+        [
```

Comment:
`ProfileLimitedSerializer` is a `ModelSerializer` and we have a dict here because I was trying to keep indexing time down by reusing the computed values out of `serialize_program_enrolled_user`. To be sure, I gave switching a try again and ended up with this exception:

```
    def get_username(self, obj):
        """Getter for the username field"""
>       return get_social_username(obj.user)
E       AttributeError: 'dict' object has no attribute 'user'
```
Which explains some code you [commented on earlier](https://github.com/mitodl/micromasters/pull/2908#discussion_r108709988) that I've since removed. I think it's better to just stick with this code than monkey patch the serializer to work with dicts.